packages/live-preview-sdk/src/inspectorMode/index.ts

```diff
@@ -202,15 +207,36 @@ export class InspectorMode {
     );
 
     this.taggedElements = entries;
+    const sendTaggedElementsMessage = () => {
+      sendMessageToEditor(
+        InspectorModeEventMethods.TAGGED_ELEMENTS,
+        {
+          elements: entries.map((e) => ({
+            coordinates: e.getBoundingClientRect(),
+          })),
+        },
+        this.targetOrigin
+      );
+    };
 
-    sendMessageToEditor(
-      InspectorModeEventMethods.TAGGED_ELEMENTS,
-      {
-        elements: entries.map((e) => ({
-          coordinates: e.getBoundingClientRect(),
-        })),
-      },
-      this.targetOrigin
-    );
+    const mutationObserver = new MutationObserver(sendTaggedElementsMessage);
+
+    this.taggedElements.forEach((element) => {
+      mutationObserver.observe(element, {
+        attributes: true,
+        attributeFilter: [
+          InspectorModeDataAttributes.ENTRY_ID,
+          InspectorModeDataAttributes.FIELD_ID,
+          InspectorModeDataAttributes.LOCALE,
+        ],
+        childList: true,
+        subtree: true,
+        characterData: true,
+      });
+    });
+
+    sendTaggedElementsMessage();
+
+    return () => mutationObserver.disconnect();
```

Comment:
I think this cleanup function will never be called :(

Reply:
Ah right because this isn't added as a event listener callback...